pubsub/rocketmq/metadata.go

```diff
@@ -14,63 +14,157 @@ limitations under the License.
 package rocketmq
 
 import (
-	"errors"
 	"fmt"
 
 	"github.com/dapr/components-contrib/pubsub"
 	"github.com/dapr/kit/config"
 )
 
-var (
-	ErrRocketmqPublishMsg         = errors.New("rocketmq publish msg error")
-	ErrRocketmqValidPublishMsgTyp = errors.New("rocketmq publish msg error, invalid msg type")
-)
-
 const (
 	metadataRocketmqTag           = "rocketmq-tag"
 	metadataRocketmqKey           = "rocketmq-key"
 	metadataRocketmqShardingKey   = "rocketmq-shardingkey"
+	metadataRocketmqQueue         = "rocketmq-queue"
 	metadataRocketmqConsumerGroup = "rocketmq-consumerGroup"
 	metadataRocketmqType          = "rocketmq-sub-type"
 	metadataRocketmqExpression    = "rocketmq-sub-expression"
 	metadataRocketmqBrokerName    = "rocketmq-broker-name"
+	metadataRocketmqQueueID       = "rocketmq-queue-id"
+)
+
+type QueueSelectorType string
+
+const (
+	HashQueueSelector       QueueSelectorType = "hash"
+	RandomQueueSelector     QueueSelectorType = "random"
+	ManualQueueSelector     QueueSelectorType = "manual"
+	RoundRobinQueueSelector QueueSelectorType = "roundRobin"
+	DaprQueueSelector       QueueSelectorType = "dapr"
 )
 
+// RocketMQ Go Client Options
 type rocketMQMetaData struct {
-	AccessProto string `mapstructure:"accessProto"`
-	// rocketmq Credentials
-	AccessKey  string `mapstructure:"accessKey"`
-	SecretKey  string `mapstructure:"secretKey"`
-	NameServer string `mapstructure:"nameServer"`
-	GroupName  string `mapstructure:"groupName"`
-	NameSpace  string `mapstructure:"nameSpace"`
-	// consumer group rocketmq's subscribers
-	ConsumerGroup     string `mapstructure:"consumerGroup"`
-	ConsumerBatchSize int    `mapstructure:"consumerBatchSize"`
-	// rocketmq's name server domain
-	NameServerDomain string `mapstructure:"nameServerDomain"`
-	// msg's content-type
-	ContentType string `mapstructure:"content-type"`
-	// retry times to connect rocketmq's broker
-	Retries     int `mapstructure:"retries"`
-	SendTimeOut int `mapstructure:"sendTimeOut"`
-}
+	InstanceName      string `mapstructure:"instanceName"`
+	ConsumerGroupName string `mapstructure:"consumerGroupName"`
```

Comment:
Can you allow `consumerGroup` too for backwards-compatibility?

Reply:
yes, func parseRocketMQMetaData in metadata.go has the following code：
if rMetaData.ConsumerGroupName == "" {
			rMetaData.ConsumerGroupName = metadata.Properties["consumerGroup"]
		}